### PR TITLE
Only render Crane visible items

### PR DESCRIPTION
**Pull request by guidezpl**
- Move state (destination list) for a given tab to _FrontLayer
- Use StaggeredGridView from flutter_staggered_grid_view to create the 4, 2 or 1 column layout
    - Added support for 2 column layout
- Only render Destination cards visible above the fold
- Fix padding issue for form headers
- Fix image cropping issue on mobile 

Screenshot 2020-04-21 at 19 27 24

Related #99

### CODE REVIEW

File: lib/studies/crane/backdrop.dart [hunk 1]
```diff
@@ -54,18 +93,27 @@ class _FrontLayer extends StatelessWidget {
...
+              if (index == 0) {
+                return _header();
+              } else {
+                return DestinationCard(destination: destinations[index]);
+              }
+            },
+            staggeredTileBuilder: (index) => index == 0
+                ? StaggeredTile.fit(crossAxisCount)
+                : const StaggeredTile.fit(1),
+            itemCount: destinations.length,
```
Comment (rami-a) [inline lib/studies/crane/backdrop.dart:116]: Shouldn't this be +1 for the header?

File: lib/studies/crane/header_form.dart [hunk 2]
```diff
@@ -11,11 +11,13 @@ const appPaddingLarge = 120.0;
...
 
 class HeaderFormField {
+  final int index;
   final IconData iconData;
   final String title;
   final TextEditingController textController;
 
-  const HeaderFormField({this.iconData, this.title, this.textController});
+  const HeaderFormField(
+      {this.index, this.iconData, this.title, this.textController});
```
Comment (rami-a) [inline lib/studies/crane/header_form.dart:20]: Nit: trailing comma

File: lib/studies/crane/item_cards.dart [hunk 3]
```diff
@@ -8,114 +8,99 @@ import 'package:flutter/rendering.dart';
...
+      highlightColor: Colors.red.withOpacity(0.5),
+      onPressed: () {},
+      child: card,
+    );
   }
 }
 
-class _DestinationCard extends StatelessWidget {
-  _DestinationCard({@required this.destination}) : assert(destination != null);
+class DestinationImage extends StatelessWidget {
```
Comment (rami-a) [inline lib/studies/crane/item_cards.dart:88]: Make private I think

File: lib/studies/crane/backdrop.dart (hunk 1 above)
Comment (guidezpl) [reply lib/studies/crane/backdrop.dart:116]: Good find!

Comment (guidezpl) [reply lib/studies/crane/backdrop.dart]: It didn't work well, with it, not sure why, but keeping as is

**Comment by guidezpl** [comment]
> LGTM, does this improve framerate when switching tabs?

It's a slight improvement as the very least, moderate if the screen is smaller and fewer destination cards have to load.

**Pull request merged**